server/src/main/java/org/opensearch/indices/replication/SegmentReplicationSourceService.java

```diff
@@ -23,40 +32,43 @@
 import org.opensearch.transport.TransportService;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Service class that handles segment replication requests from replica shards.
  * Typically, the "source" is a primary shard. This code executes on the source node.
  *
  * @opensearch.internal
  */
-public class SegmentReplicationSourceService {
+public final class SegmentReplicationSourceService extends AbstractLifecycleComponent implements ClusterStateListener, IndexEventListener {
 
     private static final Logger logger = LogManager.getLogger(SegmentReplicationSourceService.class);
+    private final RecoverySettings recoverySettings;
+    private final TransportService transportService;
+    private final IndicesService indicesService;
 
     /**
      * Internal actions used by the segment replication source service on the primary shard
      *
      * @opensearch.internal
      */
     public static class Actions {
+
         public static final String GET_CHECKPOINT_INFO = "internal:index/shard/replication/get_checkpoint_info";
         public static final String GET_SEGMENT_FILES = "internal:index/shard/replication/get_segment_files";
     }
 
-    private final Map<ReplicationCheckpoint, CopyState> copyStateMap;
-    private final TransportService transportService;
-    private final IndicesService indicesService;
+    private final OngoingSegmentReplications ongoingSegmentReplications;
 
     // TODO mark this as injected and bind in Node
-    public SegmentReplicationSourceService(TransportService transportService, IndicesService indicesService) {
-        copyStateMap = Collections.synchronizedMap(new HashMap<>());
+    public SegmentReplicationSourceService(
```

Comment:
May be time to address TODO? :-)

Reply:
Updated to wire this up in Node.  #https://github.com/opensearch-project/OpenSearch/pull/3540 will handle wiring up the target service.